src/components/AvatarMenu/AvatarMenu.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { Avatar, Menu, MenuItem, ListItemIcon, ListItemText, Divider } from '@mui/material';
+import type { PropsAvatarMenu } from './AvatarMenu.types';
+import { DEFAULT_AVATAR_MENU_ITEMS } from './AvatarMenu.constants';
+import { AvatarMenuRoot, AvatarContainer, TriangleIndicator } from './AvatarMenu.styles';
+import { useAvatarMenu } from './AvatarMenu.hook';
+
+export const AvatarMenu: React.FC<PropsAvatarMenu> = ({
+	avatarUrl,
+	username,
+	menuItems = DEFAULT_AVATAR_MENU_ITEMS,
+	avatarProps,
+	fullWidth = false,
+	'data-testid': testId = 'avatar-menu',
+}) => {
+	const {
+		isOpen,
+		anchorEl,
+		handleToggle,
+		handleClose,
+		menuId,
+		triggerId,
+		focusedIndex,
+		handleMenuKeyDown,
+		setFocusedIndex
+	} = useAvatarMenu();
+
+	const enabledItems = menuItems.filter(item => !item.disabled);
+	const enabledItemsCount = enabledItems.length;
+
+	return (
+		<>
+			<AvatarMenuRoot
+				id={triggerId}
+				aria-label={`Menú de usuario${username ? `: ${username}` : ''}`}
+				aria-controls={isOpen ? menuId : undefined}
+				aria-haspopup="true"
+				aria-expanded={isOpen ? 'true' : 'false'}
+				onClick={handleToggle}
+				data-testid={testId}
+				$fullWidth={fullWidth}
+				$isOpen={isOpen}
+			>
+				<AvatarContainer $fullWidth={fullWidth}>
+					<Avatar
+						src={avatarUrl}
+						alt={username ? `${username.toUpperCase()}` : ''}
+						{...avatarProps}
+					/>
+					<TriangleIndicator $isOpen={isOpen} />
+				</AvatarContainer>
+			</AvatarMenuRoot>
+
+
+			<Menu
+				id={menuId}
+				anchorEl={anchorEl}
+				open={isOpen}
+				onClose={handleClose}
+				disablePortal={false}
+				keepMounted={false}
+				disableScrollLock={true}
+				anchorOrigin={{
+					vertical: 'bottom',
+					horizontal: 'right',
+				}}
+				transformOrigin={{
+					vertical: 'top',
+					horizontal: 'right',
+				}}
+				slotProps={{
+					paper: {
+						sx: {
+							marginTop: '8px',
+							zIndex: 1300,
+						}
+					}
+				}}
+				MenuListProps={{
+					'aria-labelledby': triggerId,
+					onKeyDown: (e: React.KeyboardEvent<HTMLUListElement>) =>
+						handleMenuKeyDown(e, enabledItemsCount),
+					autoFocus: true,
+					dense: true,
+				}}
+			>
+				{menuItems.map((item, index) => {
+					const isEnabled = !item.disabled;
+					const isFocused = focusedIndex === index && isEnabled;
+
+					return (
+						<div key={item.label || index}>
+							<MenuItem
+								onClick={() => {
+									if (isEnabled) {
+										item.onClick();
+										handleClose();
+									}
+								}}
+								onMouseEnter={() => {
+									if (isEnabled) {
+										setFocusedIndex(index);
+									}
+								}}
+								disabled={item.disabled}
+								data-testid={item['data-testid']}
+								autoFocus={isFocused}
+								{...item.menuItemProps}
+								sx={{
+									...(isFocused && {
+										backgroundColor: 'action.hover',
```

Comment:
You set borderRadius and other style attributes in sx with hardcoded pixel values (borderRadius: '6px') and boxShadow string. Project guidelines recommend using theme tokens and avoiding px hardcodes.

**Sugerencia:**
```typescript
Use theme tokens in sx (e.g. borderRadius: 1 or theme.shape.borderRadius) and theme shadows (theme.shadows[1] or sx: { boxShadow: (theme) => theme.shadows[1] }). Example:
...(isFocused && {
  backgroundColor: 'action.hover',
  boxShadow: (theme) => theme.shadows[1],
  borderRadius: (theme) => theme.shape.borderRadius,
}),
```